### PR TITLE
Slideshow: avoid loading editor on page view

### LINKED ISSUE #28561: Slideshow: loads the editor JS on every page view
### Impacted plugin

Jetpack

### Quick summary

The Slideshow block currently loads the editor JS on every page view. This is a huge amount of JS, which severely affects page performance. The issue appears to have started with #27936, because `view.js` now imports from `slideshow.js`, which imports from the editor.

This is a classic case of insufficient modularity; the function that `view.js` is importing from `slideshow.js` (`paginationCustomRender`) should be split into its own module, that doesn't depend on the editor.

### Steps to reproduce

1. Create a simple page with a slideshow
2. Load the page on a browser, with DevTools open
3. Observe that a large amount of JS gets loaded, including the editor and all of its dependencies (React, moment, etc.)

### A clear and concise description of what you expected to happen.

The editor and its dependencies should not be loaded on a regular page view.

### What actually happened

The editor and its dependencies were loaded on a regular page view.

### Impact

All

### Available workarounds?

No but the platform is still usable

### Platform (Simple and/or Atomic)

_No response_

### Logs or notes

_No response_

### PR DESCRIPTION
**Pull request by sgomes**
As described in #28561,

> The Slideshow block currently loads the editor JS on every page view. This is a huge amount of JS, which severely affects page performance. The issue appears to have started with https://github.com/Automattic/jetpack/pull/27936, because `view.js` now imports from `slideshow.js`, which imports from the editor.
>
> This is a classic case of insufficient modularity; the function that `view.js` is importing from `slideshow.js` (`paginationCustomRender`) should be split into its own module, that doesn't depend on the editor.

This PR fixes the issue by moving the pagination code to its own module, instead of having it live in `slideshow.js`.

Fixes #28561

## Proposed changes:
* Move slideshow pagination code to its own module

### Other information:

- N/A Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
None; this is a simple bugfix.

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
1. Create a simple page with a slideshow
2. Load the page on a browser, with DevTools open
3. Observe that the editor JS does not get loaded

### CODE REVIEW

**Review by dhasilva — DISMISSED** [review]
`react` and `moment` are not present at the network requests anymore. LGTM!

**Comment by sgomes** [comment]
Thank you for the review, @dhasilva! 👍

**Comment by sgomes** [comment]
Looks like I still need an approval before I'm able to merge this. Any idea who I should ask, @dhasilva?

**Review by dhasilva — APPROVED** [review]

**Comment by dhasilva** [comment]
> Looks like I still need an approval

It is because you rebased after the approval:
```
sgomes dismissed dhasilva's stale review
```

 If you merge trunk instead you don't lose previous approvals :+1:

**Comment by sgomes** [comment]
Thanks again, @dhasilva! 👍

**Pull request merged**
